### DISCUSSION TITLE
UX Unification

### DISCUSSION THREAD

**Discussion by allcaps** [wagtail/wagtail#8158] [Ideas]
Edit: this project has been renamed to UX Unification, and will be going ahead outside of GSoC 2022 but alongside other GSoC projects.

# Apply new page editor UX to all of Wagtail

Discussion about [Apply new page editor UX to all of Wagtail](https://github.com/wagtail/wagtail/wiki/Google-Summer-of-Code-2022#apply-new-page-editor-ux-to-all-of-wagtail) a Google Summer of Code project.

> This project idea is a follow-up to Wagtail’s [Page editor 2022](https://github.com/wagtail/wagtail/discussions/7739) project, as part of which we are modernizing Wagtail’s page editor. This redesign focuses on the page editor, incorporating long-awaited UX improvements, accessibility fixes, and new features. We now want to roll out similar improvements to other parts of Wagtail.

## Reference designs

To get a sense of UX/design changes, here is a list of resources.

### Useful links

- Check out the current `main` branch, update the frontend build (`npm run build`) and run Wagtail locally, you will see a bunch of differences in the sidebar (main menu) and parts of the Page Editor
- Watch the [What's new in Wagtail 5](https://wagtail.org/blog/whats-new-in-wagtail-episode-5/)
- Review the new [page editor walk through](https://www.loom.com/share/3f9c14c576f54f4f9c77186894c9d117)

## UI mock images

Page history:

![page-history](https://user-images.githubusercontent.com/877585/162323882-170cda9c-cb96-4bfb-b73d-5148fb2dd21f.png)

Search results:

![search](https://user-images.githubusercontent.com/877585/162323936-cf2b457f-0362-4371-94aa-10f3e48be798.png)

## FAQs

1. **Does this involve the design of the new page editor?**
  - **No** - this is a follow on from the new page editor design implementation, adopting the conventions/designs and additional Figma references to other parts of Wagtail
2. **Does this involve creating new designs?**
  - **No** - the work to be done will be derived from prepared designs from a UX design team at Torchbox (the core organisation behind Wagtail) but they will likely not cover every single facet of Wagtail's UI, there will also be the start of a design system to be used as a reference.
3. **What CSS stack is used**
  - SCSS with [BEM](http://getbem.com/naming/) + [ITCSS](https://www.xfive.co/blog/itcss-scalable-maintainable-css-architecture/) classes and [Tailwind](https://tailwindcss.com/), but as this is a long lived codebase there are parts that may not match the desired new conventions. We are hoping this project will continue the momentum of our desired styling system.
4. **What is the scope of the project**
  - Please read the proposed project scope https://github.com/wagtail/wagtail/wiki/Google-Summer-of-Code-2022#apply-new-page-editor-ux-to-all-of-wagtail
  - The scope is flexible, we do not want to be prescriptive here, if a candidate wants to focus only on CSS/SCSS - that is fine and there will be more than enough to cover the project size in this alone. If the candidate wants to cover a wider area of accessibility/Django templates and JavaScript that is also fine and we will just not cover as much of the overall UI.
  - Start by looking at the new page editor designs (links above) and then compare that to other parts of Wagtail to get a feel for what may be needed to 'align' this future and old look.
  - We also have a rough spreadsheet that contains a list of all views and components https://docs.google.com/spreadsheets/d/134icHJUzr-0q42cvWhYGXtFN77mnauOT5x32sAKpW_Q/edit#gid=1433484179

## Related issues

A non-complete list of potentially related existing issues (bugs/enhancements).

Some of these may already be in the scope of the page editor work, however, it might be a good reference for those wishing to contribute or for proposals.

* https://github.com/wagtail/wagtail/discussions/8325
* https://github.com/wagtail/wagtail/issues/3236
* https://github.com/wagtail/wagtail/issues/4430
* https://github.com/wagtail/wagtail/issues/8043
* https://github.com/wagtail/wagtail/issues/8125
* https://github.com/wagtail/wagtail/issues/8134
* https://github.com/wagtail/wagtail/issues/7519
* https://github.com/wagtail/wagtail/issues/7210
* https://github.com/wagtail/wagtail/issues/4114
* https://github.com/wagtail/wagtail/issues/7366
* https://github.com/wagtail/wagtail/issues/7190
* https://github.com/wagtail/wagtail/issues/5125
* https://github.com/wagtail/wagtail/issues/434
* https://github.com/wagtail/wagtail/issues/6107
* https://github.com/wagtail/wagtail/issues/5370
* https://github.com/wagtail/wagtail/issues/4055
* https://github.com/wagtail/wagtail/issues/382
* https://github.com/wagtail/wagtail/issues/8136
* https://github.com/wagtail/wagtail/issues/4700
* https://github.com/wagtail/wagtail/issues/4058

**Comment by rishits321** [comment]
Hello, I am Rishikesh Suryavanshi, a second-year engineering undergrad student from Pune Institute Of Computer Technology, currently pursuing a bachelor's degree in Information Technology. 

I have a decent knowledge of web-dev which includes Javascript, HTML, CSS, Bootstrap, React, Jquery as well as MERN stack, and I am also familiar with Django and NumPy in Python.

I am getting used to wagtail right now and exploring its different features, but among all the project ideas, this one seems the most suitable for me.
I would like to connect with mentors to solve my doubts and to know more about this organization and this project. 
I am very excited to contribute to wagtail and would love to work on this project for GSoC 2022.

**Reply by thibaudcolas** [reply]
Hi @rishits321, welcome. Please use the [#gsoc channel on Slack](https://github.com/wagtail/wagtail/wiki/Slack) for introductions, and only post here if you have specific questions or comments about this project.

**Comment by pawan941394** [comment]
Hello Sir, My name is Pawan Kumar from India, a Third-year engineering student from Arya college of engineering and information technology.
I have excellent knowledge of web development which includes JavaScript, HTML, CSS, Bootstrap, React, jQuery, Django. I have done many full-stack web development projects in the Django framework.
I am a perfect fit for this project because I have the complete skill set that is required for this project. and if required other skills for this project then I will learn. 
I have recently completed a Microsoft virtual internship where I submitted my web project.
I would like to connect with mentors to solve my doubts and to know more about this project.
I am very excited to contribute to our organization and would love to work on this project for GSoC 2022.
I have prepared a proposal for this project you can check it here = https://drive.google.com/file/d/1Ty-Wsb1tnQL0poFCLH0kFym3HwZ2Wk9m/view?usp=sharing

**Reply by thibaudcolas** [reply]
Hi @pawan941394, welcome. Please use the [#gsoc channel on Slack](https://github.com/wagtail/wagtail/wiki/Slack) for introductions, and only post here if you have specific questions or comments about this project.

I have reviewed your application , I think it’s a bit early for applications considering how little details we’ve shared about this project so far – there are no new pages to create, and we’re not expecting contributors to do a redesign. Just implement new designs onto existing pages. This is already plenty of work as it is! For now if you want to make a solid proposal for this project I would suggest asking questions based on what you’ve found out so far.

**Comment by thibaudcolas** [comment]
For people interested in this project, I would recommend looking at:

- Our weekly updates on the [Page editor 2022](https://github.com/wagtail/wagtail/discussions/7739) project, which have a lot of background info and reference links.
- The [Wagtail UI overview](https://docs.google.com/spreadsheets/d/134icHJUzr-0q42cvWhYGXtFN77mnauOT5x32sAKpW_Q/edit) spreadsheets, which summarise all the relevant parts of Wagtail to consider.

Questions are very welcome on all of the above – this project would be invaluable for us to see through, but I realise the scope is still a bit more nebulous than some of the other ideas.

**Reply by thibaudcolas** [reply]
Quick follow-up based on yesterday’s meeting – the "Wagtail UI overview" spreadsheet is there as a planning tool. It summarises what Wagtail’s admin interface is made up of: view-by-view, and component-by-component. I expect this will help to understand the scale of this project, and plan accordingly.

**Comment by thibaudcolas** [comment]
Here is a very practical example to illustrate what this project is about :) The second most important part of the Wagtail admin is the "explorer" views – how we navigate around a site’s pages as we manage the site’s contents. Here is a screenshot of our (WIP) redesign for this part of Wagtail:

Edit: see examples here: https://github.com/wagtail/wagtail/discussions/8158#discussioncomment-2526825

Here is what this UI looks like at the moment: https://docs.wagtail.org/en/stable/editor_manual/finding_your_way_around/the_explorer_page.html

This is a good example because:

- It’s likely the part of Wagtail we’d focus on first as part of this GSoC project plan
- It’s a direct follow-up to the [page editor 2022](https://github.com/wagtail/wagtail/discussions/7739) project
- It contains components that will have already been implemented (breadcrumb, date pickers, tabs), and components that will have to be rebuilt with this new design (table’s header row)

**Comment by thibaudcolas** [comment]
Notes from Anuja’s questions in today’s meeting:

- Good place to start: Styleguide https://docs.wagtail.org/en/stable/contributing/styleguide.html#styleguide, with a list of a lot of our UI components.
- Scope: Django changes vs. front-end only? Very adaptable depending on the skills of the student.

**Reply by lb-** [reply]
Updated main description with FAQ to answer this at the top.

**Comment by lb-** [comment]
@thibaudcolas  is it possible to get links to all current Figma files (or even screen grabs), I know some are still being roughly done internally at Torchbox but some of the potential candidates are asking, plus would be good for me to get my head around if possible.


cc @PaarthAgarwal

**Reply by Dharma-09** [reply]
Yes, It will be very helpful for making good proposal and where to start :).

**Reply by lb-** [reply]
Updated the parent description with some images that have been floating around

**Reply by thibaudcolas** [reply]
Here are higher-res screenshots of three pages we have early designs of. Note this is very much experiments at this stage, that haven’t been reviewed internally and aren’t up-to-date with design changes being rolled out elsewhere.

## Page history

![page-history](https://user-images.githubusercontent.com/877585/162323882-170cda9c-cb96-4bfb-b73d-5148fb2dd21f.png)

## Search results

![search](https://user-images.githubusercontent.com/877585/162323936-cf2b457f-0362-4371-94aa-10f3e48be798.png)

**Comment by sumit-158** [comment]
Hi, everyone I have question regarding this project, is this project about implementing all these features that came up in "page editor 2022" or implementing that UX design to other page like snippets, reports , setting etc.

**Reply by lb-** [reply]
This project is about implementation of the design and potentially shared components/ templates from the new page editor to other areas of Wagtail. 

As mentioned - reports, snippets etc.

**Reply by lb-** [reply]
Updated main description with FAQ to answer this at the top.

**Comment by PaarthAgarwal** [comment]
I don't see checkboxes on the left side of page names in the mock images shared above(page hierarchy). Should they be there? If not then how will bulk actions work.

**Reply by lb-** [reply]
Great question, we will either go for showing them on hover or have them always visible, I think that will be determined as we start to work through how to split up the designs into work to be implemented.

**Reply by PaarthAgarwal** [reply]
Hover should be nice. Like initially when we hover on page name than the checkbox appears for it only but once we check it then all of them become visible.

**Reply by lb-** [reply]
yeah, we need to consider mobile devices here but that is the general gist.
Have a look at how bulk actions is implemented in page listings now to get a sense of how it works currently, we would likely not change too much of the implementation here where possible.

**Reply by thibaudcolas** [reply]
Good question indeed. What you’re suggesting makes sense to me. This is exactly the type of work we’ll need to do early on in this project – check the intended designs cover all desired functionality, and if not, understand whether any gaps are intentional (if so, look into the consequences) or not (then we need to do more designing).

**Comment by PaarthAgarwal** [comment]
Where will typescript and react come in handy during this project? While writing tests?

**Reply by lb-** [reply]
Some components, not many, are written in React and most of those are using TypeScript, so if we need to adjust the styles of those components (or potentially behaviour) it will help to know React but this is not critical.

Unit tests however are written in JavaScript, for non-React components we also try to write solid unit tests for these. We use Jest for unit tests, we also have Integration tests https://docs.wagtail.org/en/stable/contributing/developing.html#integration-tests

**Comment by ajitesh13** [comment]
Hello @lb- @helenb @thibaudcolas, I am Ajitesh, a software engineer at HackerRank. I mostly work on React with JavaScript/TypeScript and am open to exploring Django. Recently we have been through an accessibility audit applying for [WCAG 2.1 AA](https://github.com/wagtail/wagtail/projects/5) so, I have got a great experience in Accessible UI development and it seems I could be a good fit for the project, can I open a proposal for this project? Or is it late to develop a proposal ? Thank you!

**Reply by lb-** [reply]
@Ajitesh13 thanks for your interest. Please use the slack channel #gosc
https://github.com/wagtail/wagtail/wiki/Slack

For introductions and I recommend you read Wagtail's wiki page and Google Summer of Code's page for details. Proposals can be submitted from this week until the 20th of April

This discussion is for specific questions about the project and as such I will close this discussion item. 

https://github.com/wagtail/wagtail/wiki/Google-Summer-of-Code-2022#apply-new-page-editor-ux-to-all-of-wagtail

**Comment by thibaudcolas** [comment]
FYI I have removed screenshots of the "page explorer" mockups above because we don’t want people to get held up on that specific design. This was a very early draft of a highly complex UI, other screenshots above are also early draft, so there is no point in discussing the specific functionality of any of them at this stage. I’ve only shared those as examples of our design vision for parts of Wagtail that aren’t the [page editor](https://github.com/wagtail/wagtail/discussions/7739).

**Comment by lb-** [comment]
Good question came up in Slack about testing, here is what I wrote.

Hey, in regards to testing, we have
* unit tests (written in Jest) https://docs.wagtail.org/en/latest/contributing/developing.html#unit-tests-for-javascript
* integration tests (pretty minimal, uses Puppeteer) https://docs.wagtail.org/en/latest/contributing/developing.html#integration-tests
* pattern library using Storybook https://docs.wagtail.org/en/latest/contributing/developing.html#using-the-pattern-library
* manual testing on specific changes in our supported browsers https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support

For more resources
* I recommend you read about the 'Testing Pyramid' https://martinfowler.com/articles/practical-test-pyramid.html (or google that term for more articles)
* Look at the Sidebar code for a React example of all of this (mostly) done as we want https://github.com/wagtail/wagtail/tree/main/client/src/components/Sidebar 
* Have a look at how tabs are built/managed in the main code - start here client/src/includes/tabs.js & here wagtail/admin/templates/wagtailadmin/panels/tabbed_interface.html for getting a feel for this code

**Comment by lb-** [comment]
Added https://github.com/wagtail/wagtail/discussions/8325 to the linked issues/discussion mainly as a reference of potentially related improvements to the dashboard.

**Comment by lb-** [comment]
Update - I have  reviewed all the apply page editor UX proposals today, we will likely to another round of reviews soon.
[Note: also posted this in slack.](https://wagtailcms.slack.com/archives/CTL4SFX29/p1649845110422289?thread_ts=1649762289.565839&cid=CTL4SFX29)

In general
* Be super clear about how you intend to tackle this as it is lots of small changes across the whole admin, think about how you want to break up the work at a view/component level.
* Ensure you remember that the page editor work is out of scope, this is being done as part of the different Page Editor 2022 work. So no need to include that in your proposal (this includes the sidebar, page editor fields, chooser modals mostly).
* Remember there is a release planned for the middle of the project. When we merge things into main they will ned up in the next release, so you need to think about blocks of work that can be done, merged and ready for the next release instead of a 12 week long 'draft PR' that is huge and comes in at the end.
* We will have guiding designs from the Torchbox UX team, but these will not be high fidelity and we will expect many of the changes can be done without needing to granularly design every single pixel first.
* Ensure you understand the basics of accessibility and what our goal of AA compliance actually means.

**Comment by lb-** [comment]
BETA version of the Wagtail design system (for the community)
https://www.figma.com/community/file/1098619062654073532

**Comment by lb-** [comment]
For anyone watching this discussion, there is now a project board - still early days though.

https://github.com/orgs/wagtail/projects/3/views/1

**Comment by davidjeba** [comment]
Proposing to improvise the frontend editing more than backend editing

something better than
kepler.app
wix.com
nimbuseditor.com

**Reply by lb-** [reply]
@davidjeba thanks for the ideas but this discussion is more about aligning the new user interface styling / UX to other parts of Wagtail.

I saw you commented on the page editor UI item and do not want to bounce you around but I recommend you try to frame up your thoughts in what you would like to see Wagtail implement.

Then start a new discussion about this and see what feedback you get, do not be disheartened if you do not get much and also remember that anyone can build a package (yourself included) that achieves your goal.

I think our current StreamField approach is quite powerful and maybe could be extended to do more drag/drop type things in the future but I am not aware of that being on the roadmap and it would be a huge undertaking.

In the new discussion, maybe have a think about how to frame up what you are proposing.

* What does Wagtail do well right now
* What use cases does Wagtail's current UI not really serve well
* What are some examples of the goal (the links you put above)
* How might these be implemented from a technical view (if you have ideas - that would be great)

I think that a complete change to the paradigm that Wagtail's UI uses is probably going to be a hard thing to get done (as most developers in Wagtail are volunteers or do big features via sponsorship). However, it is still worth discussing and seeing who else may be interested in contributing or what is already out there in the community.

Thanks for being involved in the community

**Comment by lb-** [comment]
Update - two main issues will be the next big ones to tackle, still refining but posting here for anyone watching to add thoughts.

* https://github.com/wagtail/wagtail/issues/8645 (breadcrumbs)
* https://github.com/wagtail/wagtail/issues/8539 (header)

These two are intertwined so depending on designs received we may do parts in parallel, but most likely will try to look at the bulk of the breadcrumbs first.

**Comment by lb-** [comment]
Update - getting close to the Release Candidate 4.0 and it is incredible so see the amazing work as a whole that @PaarthAgarwal  has done.

See https://github.com/wagtail/wagtail/pull/8943/files for a bit of a preview (also, this is not even all of it, just the things that group well under the UX Unification banner).

----



 * Breadcrumbs
   * Add Breadcrumbs to the Wagtail pattern library
   * Enhance new breadcrumbs so they can be added to any header or container element
   * Adopt new breadcrumbs on the page explorer (listing) view and the page chooser modal, remove legacy breadcrumbs code for move page as no longer used
   * Rename `explorer_breadcrumb` template tag to `breadcrumbs` as it is now used in multiple locations
   * Remove legacy (non-next) breadcrumbs no longer used, remove `ModelAdmin` usage of breadcrumbs completely and adopt consistent 'back' link approach
 * Headers
   * Update classes and styles for the shared header templates to align with UI guidelines
   * Switch all report workflow, redirects, form submissions, site settings views to use Wagtail’s reusable header component
   * Resolve issues throughout Wagtail where the sidebar toggle would overlap content on small devices
 * Tabs
   * Add Tabs to the Wagtail pattern library
   * Adopt new Tabs component in the workflow history report page, removing the bespoke implementation there
 * Dashboard (home) view
   * Migrate the dashboard (home) view header to the shared header template and update designs
   * Refresh designs for Home (Dashboard) site summary panels, use theme spacing and colours
   * Add support for RTL layouts and better support for small devices
   * Add CSS support for more than three summary items if added via customisations
 * Page Listing view
   * Adopt the slim header in page listing views, with buttons moved under the "Actions" dropdown.
   * Add convenient access to the translate page button in the parent "more" button

**Comment by PaarthAgarwal** [comment]
Prepared a list of screenshots of the pages and components that I, along with my mentors, worked on during this project to compare Wagtail 3.x with the new release candidate 4.x.
Link to screenshots: https://1drv.ms/u/s!Ai2a2L22d_p2gcoScyaYsLRxGFu5bQ?e=Ic6qpq.

**Reply by lb-** [reply]
| | **3.x** | **4.x** |
|-|----|----|
| Sign in | ![Login Page 3 x](https://user-images.githubusercontent.com/1396140/187173508-483b10da-d460-4357-973f-01559215ee62.png) | ![Login Page 4 x](https://user-images.githubusercontent.com/1396140/187173572-68abaf89-49f9-49d7-9fe9-6a144ad9d677.png) |
| Dashboard | ![Dashboard 3 x](https://user-images.githubusercontent.com/1396140/187173674-8db56758-acec-43da-8364-b61187029c19.png) | ![Dashboard 4 x](https://user-images.githubusercontent.com/1396140/187173689-08f23659-efc8-4e61-a7e4-665882131dca.png) |
| Page listing | ![Page Listings View 3 x](https://user-images.githubusercontent.com/1396140/187173718-4f35e046-ab2e-46b3-887c-01724f313703.png) | ![Page Listings View](https://user-images.githubusercontent.com/1396140/187173738-50ca056f-d3e0-4dd2-af61-2c0b0c394637.png) |
| Form submission listing (mobile) | ![Form Submission Page 3 x mobile](https://user-images.githubusercontent.com/1396140/187173787-9197c989-d42f-4f21-bf61-66c90425c223.png) | ![Form Submissions Page 4 x mobile](https://user-images.githubusercontent.com/1396140/187173805-c3f0c1dc-c498-41a3-ad66-62bae808c643.png) |


Incredible work here @PaarthAgarwal  - here are some 1:1 comparisons.

**Comment by lb-** [comment]
## Closing highlights from this project

A final blog post should be coming out soon, but wanted to highlight some of the great wins from this work done by @PaarthAgarwal 

* Two of the oldest issues was able to be closed, from 2017 - https://github.com/wagtail/wagtail/issues/3823 & 2016 - https://github.com/wagtail/wagtail/issues/2309
* Two old PRs were rebased and cleaned up ready to be merged in - https://github.com/wagtail/wagtail/pull/5785 & https://github.com/wagtail/wagtail/issues/5176
* Paarth is now the [21st top contributor to Wagtail](https://github.com/wagtail/wagtail/graphs/contributors), having not contributed at all from the start of this year. With over 55 PRs, 59 commits and 25 issues resolved.

### Key highlights

* Pattern Library (Storybook)
* Tabs clean up
* Breadcrumbs
* Headers (so many!)
* Dashboard
* Page index Listing
* Buttons
* Accessibility
* JavaScript unit tests
* Sign in page / Authentication pages
* Writing blog post(s)
* Release 4.0 bug fixes 😬
* BEM/ITCSS/Tailwind & everything in between
* Documentation improvements on the side
* Report filters & form response listings fixes

### Full resolved issues list

Note: Some were done pre GSoC starting officially.

1. https://github.com/wagtail/wagtail/issues/7624
2. https://github.com/wagtail/wagtail/issues/7994
3. https://github.com/wagtail/wagtail/issues/8121
4. https://github.com/wagtail/wagtail/issues/8135
5. https://github.com/wagtail/wagtail/issues/5176
6. https://github.com/wagtail/wagtail/issues/8016
7. https://github.com/wagtail/wagtail/issues/6886
8. https://github.com/wagtail/wagtail/issues/7558
9. https://github.com/wagtail/wagtail/issues/8512
10. https://github.com/wagtail/wagtail/issues/8593
11. https://github.com/wagtail/wagtail/issues/8631
12. https://github.com/wagtail/wagtail/issues/8644
13. https://github.com/wagtail/wagtail/issues/8645
14. https://github.com/wagtail/wagtail/issues/6904
15. https://github.com/wagtail/wagtail/issues/8539
16. https://github.com/wagtail/wagtail/issues/8791
17. https://github.com/wagtail/wagtail/issues/8868
18. https://github.com/wagtail/wagtail/issues/8818
19. https://github.com/wagtail/wagtail/issues/2309
20. https://github.com/wagtail/wagtail/issues/8942
21. https://github.com/wagtail/wagtail/issues/5410
22. https://github.com/wagtail/wagtail/issues/8992
23. https://github.com/wagtail/wagtail/pull/9043 (no issue raised)
24. https://github.com/wagtail/wagtail/issues/8790
25. https://github.com/wagtail/wagtail/issues/9053
26. https://github.com/wagtail/wagtail/issues/9014
27. https://github.com/wagtail/wagtail/issues/8661
28. https://github.com/wagtail/wagtail/issues/9083
29. https://github.com/wagtail/wagtail/issues/3823
30. https://github.com/wagtail/wagtail/issues/3236
31. https://github.com/wagtail/wagtail/issues/8043

#### Items outside the main Wagtail repo

1. https://github.com/wagtail/stylelint-config-wagtail/issues/14
2. https://github.com/wagtail/stylelint-config-wagtail/issues/15
3. https://github.com/wagtail/sphinx_wagtail_theme/issues/176
4. https://github.com/wagtail/sphinx_wagtail_theme/issues/170
5. https://github.com/wagtail/sphinx_wagtail_theme/issues/56
6. https://github.com/wagtail/sphinx_wagtail_theme/issues/9
7. https://github.com/wagtail/sphinx_wagtail_theme/issues/161
8. https://github.com/wagtail/sphinx_wagtail_theme/issues/164
9. https://github.com/wagtail/sphinx_wagtail_theme/issues/20
10. https://github.com/wagtail/sphinx_wagtail_theme/issues/80
11. https://github.com/wagtail/sphinx_wagtail_theme/issues/22
12. https://github.com/wagtail/sphinx_wagtail_theme/issues/61
13. https://github.com/wagtail/sphinx_wagtail_theme/issues/131


## Future issues

There were of course many things that did not get done or were split out / discovered during this project and anyone is welcome to help contribute to these to make Wagtail even better.

See https://github.com/wagtail/wagtail/issues/9228#issuecomment-1250905926 for the list.
